README.md

```diff
@@ -1,28 +1,52 @@
-4. Add initial app description to README and update initial required ENV variable documentation as needed.
-5. Update license if needed (check app-specific dependencies for licensing terms).
-6. Check Github repository settings:
-   - Confirm repo branch protection settings are correct (see [dev docs](https://mitlibraries.github.io/guides/basics/github.html) for details)
-   - Confirm that all of the following are enabled in the repo's code security and analysis settings:
-      - Dependabot alerts
-      - Dependabot security updates
-      - Secret scanning
-7. Create a Sentry project for the app if needed (we want this for most apps):
-   - Send initial exceptions to Sentry project for dev, stage, and prod environments to create them.
-   - Create an alert for the prod environment only, with notifications sent to the appropriate team(s).
-   - If *not* using Sentry, delete Sentry configuration from config.py and test_config.py, and remove sentry_sdk from project dependencies.
+# transmogrifier-ab-diff
+
+Compare transformed TIMDEX records from two versions (A,B) of Transmogrifier.
+
+# abdiff
+
+`abdiff` is the name of the CLI application in this repository that performs an A/B test of Transmogrifier by 
+performing the following:
+
+1. initialize a new "Job" with two git commit SHAs of Transmogrifier (A and B) to compare the transformed records from
+2. locally build A and B Docker images of Transmogrifier based on these commits
+3. accept and transform a batch of input files, for A and B versions of Transmogrifier, constituting a "Run"
+4. collate all A and B transformed records into a single parquet file
+5. for each `timdex_record_id` create a diff of version A and B
+6. generate metrics derived from all diffs
+7. provide a Flask app to view and explore the metrics
+
+## Job, Run, and File Structure
+
+A "Job" is defined as two versions of Transmogrifier to compare.  A "Run" is a run of this Job, given a set of input files
+to transform.  Each Job gets its own directory, and each Run is fully encapsulated in a sub-directory of the Job.  All of this
+is written to local `./output` directory.
+
+This results in a file structure similar to the following:
+
+```text
+├── output
+│   └── test-refactor-1
+│       ├── job.json
+│       └── runs
+│           ├── 2024-08-23_12-10-00
+│           │   ├── collate_ab.parquet
+│           │   ├── diff_ab.parquet
+│           │   ├── metrics_ab.json
+│           │   ├── run.json
+│           │   └── transformed
+│           │       ├── a
+│           │       │   ├── alma-2024-01-01-transformed-to-index.json
+│           │       │   └── dspace-2024-03-15-transformed-to-index.json
+│           │       └── b
+│           │           ├── alma-2024-01-01-transformed-to-index.json
+│           │           └── dspace-2024-03-15-transformed-to-index.json
+│           └── 2024-08-23_13-30-00
+                └── # and similar structure here for this run...
+```
```

Comment:
Hmm, how do you feel about rewriting this section to:


## Concepts

1. **Job**: A job in `abdiff` represents the A/B test for comparing the results from two versions of Transmogrifier. When a job is first created, a JSON file with an initial set of configurations is created in the root of the job folder (`job.json`). These configurations are described below:

   ```shell
   {
     "job_name"   <------- command `job-init` writes
     "version_a_sha"   <------ command 'job-init' writes this
     "version_b_sha"   <------ command 'job-init' writes this
   }
   ```

2. **Run**: A run is the _execution_ of a job. The outputs from a run are fully encapsulated in a nested sub-directory of the job folder, with each run uniquely identified by the timestamp of execution (formatted as X). When a run is executed, the job JSON file is copied into the run folder. Over the course of the run, the copied JSON file (`run.json`) is updated with X.

    This results in a file structure similar to the following:

    ```text
    ├── output
    │   └── test-refactor-1
    │       ├── job.json
    │       └── runs
    │           ├── 2024-08-23_12-10-00
    │           │   ├── collate_ab.parquet
    │           │   ├── diff_ab.parquet
    │           │   ├── metrics_ab.json
    │           │   ├── run.json
    │           │   └── transformed
    │           │       ├── a
    │           │       │   ├── alma-2024-01-01-transformed-to-index.json
    │           │       │   └── dspace-2024-03-15-transformed-to-index.json
    │           │       └── b
    │           │           ├── alma-2024-01-01-transformed-to-index.json
    │           │           └── dspace-2024-03-15-transformed-to-index.json
    │           └── 2024-08-23_13-30-00
                   └── # and similar structure here for this run...
    ```

---

**Note:** For instance, I could see us eventually adding a third item like "Diff" to describe what changes the CLI actually captures. Could also see some of this moving to its own doc under docs eventually but keeping things in the README is good for now. Just thought it'd be nice to move some of what we've decided/discussed (from Jira tickets) into actual docs!

Reply:
I like it!  Thanks for suggesting.  I've taken this, and reworked slightly in [this commit](https://github.com/MITLibraries/transmogrifier-ab-diff/pull/1/commits/0d4a141c146c69b007dcd879f719d17f1b397732).

Reply:
Awesome! :D